morpho/processors/sampling/PriorSamplingProcessor.py

```diff
@@ -0,0 +1,84 @@
+'''
+Processor for sampling inputs to a data generator from priors.
+Useful for sensitivity analyses.
+
+Authors: T. E. Weiss
+Date: May 2020
+'''
+
+from __future__ import absolute_import
+
+import numpy as np
+
+from morpho.utilities import morphologging, reader
+from morpho.processors import BaseProcessor
+logger = morphologging.getLogger(__name__)
+
+__all__ = []
+__all__.append(__name__)
+
+
+class PriorSamplingProcessor(BaseProcessor):
+    '''
+    Input:
+        priors: list of dictionaries with the following required keys:
+            'name' - str; parameter name
+            'prior_dist' - str; name of prior distribution to sample from (e.g. 'gamma')
+            'prior_params' - list of values parameterizing prior, following numpy conventions (e.g. [1, 2] for beta prior with k=1 and theta=2)
+        fixed_inputs: (optional) dictionary containing any inputs to the next processor/data generator which will *not* be sampled from priors
+
+    Results:
+        sampled_inputs: dictionary containing values sampled from priors as designated in the priors dict, as well as the keys/values in the fixed_inputs dict
+    '''
+    def InternalConfigure(self,params):
+        self.priors = reader.read_param(params,'priors',{})
+        self.fixed_inputs = reader.read_param(params,'fixed_inputs',{})
+        return True
+    
+    def _sample_inputs(self):
+        sampled_inputs = self.fixed_inputs
+        
+        logger.info("Sampling values from priors")
+        for sample_dict in self.priors:
+            name = sample_dict['name']
```

Comment:
Maybe we should make sure this name doesn't exist in the sampled_inputs first? In case the user made a typo between the fixed inputs or the sampled ones...